src/java.base/share/classes/sun/security/ssl/SignatureScheme.java

```diff
@@ -413,11 +413,26 @@ private static List<SignatureScheme> getSupportedAlgorithms(
             List<ProtocolVersion> activeProtocols,
             Set<SSLScope> scopes) {
         List<SignatureScheme> supported = new LinkedList<>();
+        List<SignatureScheme> schemesToCheck;
 
-        List<SignatureScheme> schemesToCheck =
-                config.signatureSchemes == null ?
-                    Arrays.asList(SignatureScheme.values()) :
-                    namesOfAvailable(config.signatureSchemes);
+        // No need to look up the names of the default signature schemes.
+        if (config.signatureSchemes == SupportedSigSchemes.DEFAULT
+                || config.signatureSchemes == null) {
+            schemesToCheck = Arrays.asList(SignatureScheme.values());
+        } else {
+            schemesToCheck = Arrays.stream(config.signatureSchemes)
+                    .map(name -> {
+                        var ss = SignatureScheme.nameOf(name);
+                        if (ss == null && SSLLogger.isOn
+                                && SSLLogger.isOn("ssl,handshake")) {
```

Comment:
Do you need both `SSLLogger.isOn` and the isOn check for ssl,handshake?  The latter check appears to only return true if it is both on and those flags have been set.